helix-view/src/document.rs

```diff
@@ -560,8 +561,30 @@ impl Document {
                 }
             }
 
-            let mut file = File::create(path).await?;
-            to_writer(&mut file, encoding, &text).await?;
+            // TODO Temporary file creation is a workaround to solve large file corruption
+            // that occurs when crashing during a file save
+            let tmp_path = cache_dir().join(format!("~{:?}", path.file_name().unwrap()));
+
+            if let Some(parent) = tmp_path.parent() {
+                // TODO Temporary fix. Normally, this directory will exist by this point,
+                // but not during the integration tests.
+                if !parent.exists() {
+                    if force {
```

Comment:
I don't think the write test does `:w!` (force write).

Reply:
https://github.com/helix-editor/helix/blob/a3ed9169df8d0a56ab4f698d2c90e7e6b4404633/helix-term/tests/test/write.rs#L18
So it still doesn't build the directory.

Reply:
Oh, that makes sense, hence it doesn't change how the CI works

Reply:
Yeah I think it's fine to create the cache directory in `save_impl` if it doesn't exist.